src/main/scala/com/cognite/sdk/scala/v1/fdm/views/ViewPropertyCreateDefinition.scala

```diff
@@ -22,23 +25,15 @@ object ViewPropertyCreateDefinition {
   implicit val createViewPropertyDecoder: Decoder[CreateViewProperty] =
     deriveDecoder[CreateViewProperty]
 
-  final case class ConnectionDefinition(
```

Comment:
let's do
```scala
  final case class CreateConnectionDefinition(connectionDefinition: ConnectionDefinition)
    extends ViewPropertyCreateDefinition
  implicit val createConnectionDefinitionEncoder: Encoder[CreateConnectionDefinition] =
    PropertyDefinition.connectionDefinitionEncoder.contramap(_.connectionDefinition)
  implicit val createConnectionDefinitionDecoder: Decoder[CreateConnectionDefinition] =
    PropertyDefinition.connectionDefinitionDecoder.map(CreateConnectionDefinition)
```
pretty short and allows keeping the trait sealed. Only needs couple small changes to compile: viewPropertyEncoder match no longer needs `_` case, viewPropertyDecoder needs to refer to CreateConnectionDefinition, ViewsTest needs
```diff
-      properties = viewWithRDRProperties
+      properties = viewWithRDRProperties.mapValues(CreateConnectionDefinition).toMap
```

Reply:
But does that mean that anybody using the SDK to create connections will also need to to the mapValues(CreateConnectionDefinition).toMap ? Is that ok?

Reply:
also in thsi case it works to do just mapValues, but what if there are other properties than just this one? I think that's trading a bit too much clarity in the code for API clarity.

Reply:
yes, surface classes change, how much adaptation is needed may depend on the use-case, also convenience util conversions could be added if needed. This is already the case for the other property type that there's a need to shift types if for example data is read from one model and read to another, so hopefully not a big change overall

Reply:
If there are other properties then the code already deals with `ViewPropertyCreateDefinition` and no conversion will be needed

Reply:
I guess you can do:

```
    val reverseDirectRelationProperty = CreateConnectionDefinition(
        ReverseDirectRelationConnection(
        Some("name"),
        Some("desc"),
        MultiReverseDirectRelation,
        ViewReference(spaceName, viewPointedTo.externalId, viewVersion1),
        ThroughConnection("connection", ViewReference(spaceName, viewPointedTo.externalId, viewVersion1))
      )
    )
```
so it's not that bad, but I'm still not sure it's that nice from an API POV

Reply:
To make surface compatible with both types it should be easy to just one extra constructor to `ViewCreateDefinition` object that accepts `ViewPropertyCreateDefinition` non-create counterpart and does the conversion (so the `ViewPropertyDefinition` it seems)

Reply:
Ok, I asked Jaime his opinion and he agrees with you, let's go with that then!

Reply:
> To make surface compatible with both types it should be easy to just one extra constructor to `ViewCreateDefinition` object that accepts `ViewPropertyCreateDefinition` non-create counterpart and does the conversion (so the `ViewPropertyDefinition` it seems)

Not that simple right? Because the viewProperties could be a mix, so the map has to be a map of Map[String, ViewPropertyCreateDefinition | ConnectionDefinition] and then transforma all connectionDefinitions to CreateConnectionDefinitions, but scala doesn't support maps like that unless they share a trait.

Reply:
But I guess it's ok without it

Reply:
`ViewPropertyCreateDefinition | ConnectionDefinition` use case should be not very common, either Create* are created from scratch as creates already, or they come from reading the api and then they are ViewPropertyDefinition and can be converted with extra util function. ConnectionDefinition is "shared" between those 2 traits and they are just to define create/read endpoints so those 2 cases should cover the most typical scenarios :crossed_fingers:

Reply:
For the rest looks green, will have a quick final read and should be good to go :tada: